File: Algorithm_practice/sorting/quick_sort.cpp
#include <stdio.h>

int main(void) {
    int arr[10] = {3, 2, 1, 4, 7, 6, 10, 5, 8, 9};
    int pivot, temp;
    
    for(pivot=0; pivot<9; pivot++) {
        int i = pivot;
        int j = 9;

        while(i<j) {
            while(arr[pivot]>arr[i]) {
                i++;
            }
            while(arr[pivot]<arr[j]) {
                j--;
            }
            temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
        temp = arr[pivot];
        arr[pivot] = arr[i];
        arr[i] = temp;
    }
    
    for (int i = 0; i < 10; i++) {
		printf("%d ", arr[i]);
	}

    return 0;
}

File: Algorithm_practice/Algorithm_py/quick_sort.py
# 오른쪽으로 작은 놈
# 왼쪽으로 큰 놈

arr =[3, 2, 1, 4, 7, 6, 10, 5, 8, 9]

for idx in range(len(arr)):
    i, j = 0, idx
    while(i<j):
        while(arr[idx]> arr[i]):
            i+=1
        while(arr[idx]< arr[j]):
            j-=1
        temp = arr[i]
        arr[i] = arr[j]
        arr[j] = temp
    temp = arr[idx]
    arr[idx] = arr[j]
    arr[j] = temp
    
print(arr)
